gui_views/button_view.py

```diff
@@ -473,6 +476,9 @@ def save_as_UI_button_pressed(self):
         if not filename:
             return
 
+        # add .ui extension by default
+        filename += ".ui"
```

Comment:
if the user manually puts .ui will this cause weird behavior?

Reply:
I think it will just create a file with 2 ".ui"s but it would still treat the file the same.

Reply:
is that good behavior / practice tho

Reply:
it now checks if there's an extension before adding it